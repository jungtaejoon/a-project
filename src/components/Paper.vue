<template>
  <div>
    <paragraph-history
      v-for="paragraphHistory in paragraphHistories"
      :key="paragraphHistory"
      :id="paragraphHistory"
      @need-new-paragraph="createParagraphHistory"
      @delete-paragraph-history="deleteParagraphHistory"
    />
    <button @click="createParagraphHistory">단락 추가</button>
  </div>
</template>

<script>
import ParagraphHistory from './ParagraphHistory'

export default {
  name: 'paper',
  components: {
    ParagraphHistory
  },
  data () {
    return {
      paragraphHistories: [],
      nextParagraphHistoryId: 0
    }
  },
  created () {
    this.createParagraphHistory()
  },
  methods: {
    createParagraphHistory (paragraphHistoryId) {
      if (typeof paragraphHistoryId === 'number') {
        const targetIndex = this.paragraphHistories.indexOf(paragraphHistoryId)
        this.paragraphHistories.splice(targetIndex + 1, 0, this.nextParagraphHistoryId++)
      } else {
        this.paragraphHistories.push(this.nextParagraphHistoryId++)
      }
    },
    deleteParagraphHistory (paragraphHistoryId) {
      const targetIndex = this.paragraphHistories.indexOf(paragraphHistoryId)
      if (targetIndex !== -1) {
        this.paragraphHistories.splice(targetIndex, 1)
      }
    }
  }
}
</script>

<style scoped>
</style>
