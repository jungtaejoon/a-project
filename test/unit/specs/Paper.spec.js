import {mount} from '@vue/test-utils'
import Paper from '@/components/Paper'

describe('Paper.vue', () => {
  const renderer = require('vue-server-renderer').createRenderer()
  const expected = [0]
  it('초기 세팅이 성공해야 한다', () => {
    const wrap = mount(Paper)
    expect(wrap.vm.paragraphHistories).toEqual(expect.arrayContaining(expected))
    expect(wrap.vm.nextParagraphHistoryId).toEqual(1)
    renderer.renderToString(wrap.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
