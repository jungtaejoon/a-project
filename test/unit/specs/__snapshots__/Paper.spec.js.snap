// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper.vue should render correct contents 1`] = `
<div>
    <div id="paragraphHistory-0">
        <textarea name="" id="paragraph-1"></textarea>
    </div>
    <button>단락 추가</button>
</div>
`;

exports[`Paper.vue 초기 세팅이 성공해야 한다 1`] = `
<div>
    <div id="paragraphHistory-0">
        <textarea name="" id="paragraph-1"></textarea>
    </div>
    <button>단락 추가</button>
</div>
`;
